source/includes/connection-snippets/x509-snippets/cpp-connection.cpp

```diff
@@ -0,0 +1,25 @@
+// begin x509 connection
+#include <iostream>
+#include <bsoncxx/builder/stream/document.hpp>
+#include <bsoncxx/json.hpp>
+#include <mongocxx/client.hpp>
+#include <mongocxx/uri.hpp>
+#include <mongocxx/instance.hpp>
+#include <mongocxx/options/client.hpp>
+#include <mongocxx/options/ssl.hpp>
+
+int main(int, char**) {
+  mongocxx::instance inst{};
+
+  mongocxx::options::ssl ssl_opts{};
+  ssl_opts.ca_file("/etc/certs/mongodb/ca.pem");
+  ssl_opts.pem_file("/etc/certs/mongodb/client.pem");
+
+  mongocxx::options::client client_opts{};
+  client_opts.ssl_opts(ssl_opts);
+
+  mongocxx::uri uri("mongodb+srv://<your-x509-client-subject>@<cluster-url>/?authMechanism=MONGODB-X509&ssl=true");
```

Comment:
Note that C++ doesn't yet support the Unified URI options spec (per https://jira.mongodb.org/browse/DRIVERS-557), so you're correct to use `ssl_opts` here.

I do think that `ssl=true` may be redundant in the URI, though. In libmongoc, which C++ uses, SSL is enabled for a client when SSL opts are assigned, even if the connection string did not specify `ssl=true` or `tls=true` on its own.

cc: @bazile-clyde @kevinAlbs

Reply:
Agreed, `ssl=true` or `tls=true` is redundant if setting via `ssl_opts`.

Reply:
@mongomoe let's remove `ssl=true` here, and then I think this can be resolved. Doing so would also be consistent with the C driver example, since that connection string omits an `ssl` or `tls` option as well.

Reply:
Apologies, `ssl=true` is actually not redundant because the C++ driver has an [extra validation check](https://github.com/mongodb/mongo-cxx-driver/blob/fe4b1dcc2b9b4402b7f7b8bb0202bbc8a227ccf1/src/mongocxx/client.cpp#L70-L74)  which throws an exception if attempting to set SSL options without having `ssl=true` in the URI.
I'll note in the [new review](https://github.com/mongodb/docs-ecosystem/pull/583/files)